clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -291,55 +381,33 @@ bool CheckedRegionFinder::isUncheckedStruct(QualType Qt, std::set<std::string> &
   }
 }
 
+
 void CheckedRegionFinder::addUncheckedAnnotation(CompoundStmt *S, int Localwild) {
   auto Cur = S->getWrittenCheckedSpecifier();
-
   llvm::FoldingSetNodeID Id;
   S->Profile(Id, *Context, true);
-  auto Search = Seen.find(Id);
-
-  if (Search == Seen.end()) {
-    auto Loc = S->getBeginLoc();
-    bool IsChecked = !hasUncheckedParameters(S) &&
-                   Cur == CheckedScopeSpecifier::CSS_None && Localwild == 0;
 
-    Map[Id] = IsChecked ? IS_CHECKED : IS_UNCHECKED;
+  bool IsChecked = !hasUncheckedParameters(S) &&
+    Cur == CheckedScopeSpecifier::CSS_None && Localwild == 0;
 
-    // Don't add _Unchecked to top level functions.
-    if ((!IsChecked && !isFunctionBody(S))) {
-      Writer.InsertTextBefore(Loc, "_Unchecked ");
-    }
-  }
+  Map[Id] = IsChecked ? IS_CHECKED : IS_UNCHECKED;
 }
 
-bool CheckedRegionFinder::isFunctionBody(CompoundStmt *S) {
-  const auto &Parents = Context->getParents(*S);
-  if (Parents.empty()) {
-    return false;
-  }
-  return Parents[0].get<FunctionDecl>();
-}
 
 
 bool CheckedRegionFinder::VisitMemberExpr(MemberExpr *E){
```

Comment:
In terms of ordering, it seems odd to have this `visitXXX` methods here below your custom ones. I would think you'd want to group the `visitXXX` ones together, and then have `addUnchecked...` come last.